src/pools/cluster/fixed.ts

```diff
@@ -0,0 +1,161 @@
+import type { SendHandle } from 'child_process'
+import * as cluster from 'cluster'
+import { Worker } from 'cluster'
+import type { MessageValue } from '../../utility-types'
+
+export type WorkerWithMessageChannel = Worker // & Draft<MessageChannel>
+
+export interface FixedClusterPoolOptions {
+  /**
+   * A function that will listen for error event on each worker.
+   */
+  errorHandler?: (this: Worker, e: Error) => void
+  /**
+   * A function that will listen for online event on each worker.
+   */
+  onlineHandler?: (this: Worker) => void
+  /**
+   * A function that will listen for exit event on each worker.
+   */
+  exitHandler?: (this: Worker, code: number) => void
+  /**
+   * This is just to avoid not useful warnings message, is used to set `maxListeners` on event emitters (workers are event emitters).
+   *
+   * @default 1000
+   */
+  maxTasks?: number
+}
+
+/**
+ * A cluster pool with a static number of workers, is possible to execute tasks in sync or async mode as you prefer.
+ *
+ * This pool will select the worker in a round robin fashion.
+ *
+ * @author [Christopher Quadflieg](https://github.com/Shinigami92)
+ * @since 2.0.0
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export class FixedClusterPool<Data = any, Response = any> {
+  public readonly workers: WorkerWithMessageChannel[] = []
+  public nextWorker: number = 0
+
+  // workerId as key and an integer value
+  public readonly tasks: Map<WorkerWithMessageChannel, number> = new Map<
+    WorkerWithMessageChannel,
+    number
+  >()
+
+  protected id: number = 0
+
+  /**
+   * @param numWorkers Number of workers for this pool.
+   * @param filePath A file path with implementation of `ClusterWorker` class, relative path is fine.
+   * @param opts An object with possible options for example `errorHandler`, `onlineHandler`. Default: `{ maxTasks: 1000 }`
+   */
+  public constructor (
+    public readonly numWorkers: number,
+    public readonly filePath: string,
+    public readonly opts: FixedClusterPoolOptions = { maxTasks: 1000 }
+  ) {
+    if (!cluster.isMaster) {
+      throw new Error('Cannot start a cluster pool from a worker!')
+    }
+    // TODO christopher 2021-02-09: Improve this check e.g. with a pattern or blank check
+    if (!this.filePath) {
+      throw new Error('Please specify a file with a worker implementation')
+    }
+
+    cluster.setupMaster({
```

Comment:
We could want to support more options here ? https://nodejs.org/dist/latest-v14.x/docs/api/cluster.html#cluster_cluster_settings

Reply:
Could we do this in another issue/pr? I currently just want to bring the basic copy-paste of the thread to cluster transformation